rotkehlchen/db/drivers/gevent.py

```diff
@@ -366,12 +367,20 @@ def write_ctx(self, commit_ts: bool = False) -> Generator['DBCursor', None, None
                 raise
             else:
                 if commit_ts is True:
-                    cursor.execute(
-                        'INSERT OR REPLACE INTO settings(name, value) VALUES(?, ?)',
-                        ('last_write_ts', str(ts_now())),
-                    )
-                    # last_write_ts in not cached to cached settings. This is a critical section
-                    # and adding even one more function call can have very ugly and
+                    try:
+                        cursor.execute(
```

Comment:
This is not good. This is a very critical section of the app. I don't want this change here. Even the Enum.serialize() part is bad. This happens for every single DB write. This should not go here. U/ndo this change completely.

Reply:
Updated it like this:
```py
try:
    cursor.execute(
        'INSERT OR REPLACE INTO key_value_cache(name, value) VALUES(?, ?)',
        ('last_write_ts', str(ts_now())),
    )
except sqlcipher.OperationalError:  # pylint: disable=no-member
    # when key_value_cache is missing, go back to settings
    # it happens while upgrading until DB version v41
    cursor.execute(
        'INSERT OR REPLACE INTO settings(name, value) VALUES(?, ?)',
        ('last_write_ts', str(ts_now())),
    )
# last_write_ts is updated to settings/key_value_cache. This is a critical
# section and adding even one more function call can have very ugly and
# detrimental effects in the entire codebase as everything calls this.
```

The `.serialize()` function is removed, but still using try-except. Because during db upgrades and migrations in old versions, `key_value_cache` table is not present.

Reply:
this try/except happens only for the case of the upgrade right? so the usual path is just doing the insert and exiting

Reply:
Still not good. I am sorry but I don't think you realize what the critical path means.  try/excpet is not free. And having an unecessary try/except just for that 1 edge case is not acceptable.